### DISCUSSION TITLE
How to add tailwind classes on dynamically generated html content?
(Vanilla JS)

### DISCUSSION THREAD

**Discussion by Maykiyel** [tailwindlabs/tailwindcss#13463] [Help] [answered]
I'm new to tailwind css and I'm trying to figure how to work around dynamically generated content using vanilla JS. For example cards. The content does generate with the correct classes in the class list. However, the styles are not being applied. I know this is because of post css but is there a way to make this work?

**Comment by alex-krasikau** [comment]
Hey @Maykiyel!

The best approach here is to place the dynamic styles directly in the `style` attribute. Please let me know what is the technology you are using, and I will provide a demo.

**Reply by Maykiyel** [reply]
I'm using Vite with just Vanilla JS and Tailwind CSS, PostCSS, and autoprefixer installed. I tried making custom classes in my style.css stylesheet with @apply it didn't work tho, like I mentioned it's either because of post css or preflight. The approach you mentioned seems kind of tedious imo. Is that really the only way to do it? Does that mean you can't use Tailwind for stuff like this?

Thanks in advance!

**Reply by Maykiyel** [reply]
Hello again @alex-krasikau 

For context here's what I'm trying to do:
![image_2024-04-07_172728611](https://github.com/tailwindlabs/tailwindcss/assets/156945500/01de6b43-0c4f-4f6d-9bf1-8e9bb21e083f)

Then here it is now with the approach you mentioned:
![image](https://github.com/tailwindlabs/tailwindcss/assets/156945500/d02d2e18-d170-41ff-a995-1c582c226fc1)

**Reply by alex-krasikau** [reply]
I see, I didn't understand your initial goal.

Here's how I would use Tailwind CSS to build:

```javascript
...
const cardElement = document.createElement("div");
cardElement.classList.add("size-20", "perspective:1000px");

cardElement.innerHTML = `<div class="w-full h-full cursor-pointer transition-all duration-300">
  <div class="bg-[#6B451C] absolute inset-0 w-full h-full..." />
  <div class="...">
    <img src="${card.image}" alt="${card.name}" class="w-full h-full" />
  </div>
</div>`;
...
```

If it doesn't work, ensure you have correctly set up the `content` field in the Tailwind Configuration - https://tailwindcss.com/docs/content-configuration.

**Reply by Maykiyel** [reply]
Will try this next time! For now i settled with the inline styles method. When I get the time to update my project i'll try this method. Thanks a lot!

**Reply by alex-krasikau** [reply, marked as answer]
Sounds good! Feel free to create a new issue if you encounter any problems.

If Tailwind utility classes didn't style correctly before switching to inline styles, it's probably due to a wrong configuration of the `content` property in the Tailwind Configuration.

**Comment by Foerster64** [comment]
I appreciate your contribution to solving this problem. [Valero Credit Card](https://valerocreditcard.fun/) Your solution is insightful and well thought out.
